### DISCUSSION TITLE
Issue with SkiaGTK TextBox on RedHat 7

### DISCUSSION THREAD

**Discussion by unokeys** [unoplatform/uno#13948] [General]
Hello,

Creating a simple hello world app with a textbox, and running on RedHat 7.9, the TextBox has a lot of flickering issues. If you mouse over the TextBox, or type text in it, the entire screen will flicker and go black.

This is with .net 6.

We do not observe this issue on Ubuntu 22.04.

Any idea what we can do to resolve this?

Thanks

The below pics show the effect.

I am running the app via `dotnet nameOfProject.Skia.Gtk.dll`

![image](https://github.com/unoplatform/uno/assets/147554392/07f6eccf-d40d-4180-b54d-63f82a4de5d8)
![image](https://github.com/unoplatform/uno/assets/147554392/0050293d-2e86-492b-845d-83c2e6c95b6a)

Below is my MainPage.xaml as well as my .Skia.Gtk csproj files,

``` csharp
<Page
    x:Class="UnoEventHandlerTest.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:UnoEventHandlerTest"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"    
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <StackPanel Orientation="Vertical">
        <TextBox x:Name="nameInput"
                 Header="This is a standard Text Box:" PlaceholderText="Name"
                 Width="300" HorizontalAlignment="Left"
                 />
    </StackPanel>
</Page>
```

``` XML
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType Condition="'$(Configuration)'=='Release'">WinExe</OutputType>
    <OutputType Condition="'$(Configuration)'=='Debug'">Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup Condition="exists('..\UnoEventHandlerTest.Windows')">
    <EmbeddedResource Include="..\UnoEventHandlerTest.Windows\Package.appxmanifest" LogicalName="Package.appxmanifest" />
    <Content Include="..\UnoEventHandlerTest.Windows\Images\StoreLogo.png" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.WinUI.Skia.Gtk" Version="4.7.37" />
    <PackageReference Include="Uno.WinUI.RemoteControl" Version="4.7.37" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.7.37" />
    <PackageReference Include="Uno.WinUI.Lottie" Version="4.7.37" />
    <PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.3" />
    <PackageReference Include="SkiaSharp.Skottie" Version="2.88.3" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\UnoEventHandlerTest\UnoEventHandlerTest.csproj" />
  </ItemGroup>
  <Import Project="..\UnoEventHandlerTest.Base\base.props" />
</Project>
```

**Comment by jeromelaban** [comment]
Thanks for the report. 

Are you running inside a VM by any chance? If you do, it's like that the OpenGL driver is faulty in some way. You can disable [OpenGL acceleration](https://platform.uno/docs/articles/features/using-skia-gtk.html#changing-the-rendering-target) as workaround.

**Comment by unokeys** [comment]
Hi @jeromelaban . Great! That seems to fix the issue. On my real app, I intend for users to run this on both VMs and on native linux boxes

I dont fully understand all the implications of 
`host.RenderSurfaceType = RenderSurfaceType.Software;` in the GTK `static void Main()`

Is this safe to change this if I want things to work/run properly in both VM/native ?

**Reply by jeromelaban** [reply]
Thanks for the update. It's technically safe to run with the Software renderer, but it's also much slower to render the UI contents. I would suggest setting this property value when running in a VM is detected, or by using a environment variable.

**Comment by sasakrsmanovic** [comment]
@unokeys - would love to know what you are building with Uno! Let us know at info@platform.uno - we always love hearing about all the use cases we get.
